File: okex-php-sdk-api-v5/src/SubaccountApi.php
<?php


namespace okv5;


class SubaccountApi extends Utils
{
    public function balances($subAcct){
        $params = [
            'subAcct' => $subAcct,
        ];

        return $this->request('/api/v5/account/subaccount/balances', $params, 'GET');
    }

    public function bills($ccy='',$type='',$subAcct='',$after='',$before='',$limit=''){
        $params = [
            'ccy' => $ccy,
            'type' => $type,
            'subAcct' => $subAcct,
            'after' => $after,
            'before' => $before,
            'limit' => $limit,
        ];

        return $this->request('/api/v5/asset/subaccount/bills', $params, 'GET');
    }

    

    public function modifyApikey($pwd,$subAcct,$label,$apiKey,$perm,$ip=''){
        $params = [
            'pwd' => $pwd,
            'subAcct' => $subAcct,
            'label' => $label,
            'apiKey' => $apiKey,
            'perm' => $perm,
            'ip' => $ip,
        ];

        return $this->request('/api/v5/users/subaccount/modify-apikey', $params, 'POST');
    }

   

   
    public function subaccountList($enable='',$subAcct='',$after='',$before='',$limit=''){
        $params = [
            'enable' => $enable,
            'subAcct' => $subAcct,
            'after' => $after,
            'before' => $before,
            'limit' => $limit,
        ];

        return $this->request('/api/v5/users/subaccount/list', $params, 'GET');
    }

    public function transfer($ccy,$amt,$from,$to,$fromSubAccount,$toSubAccount,$loanTrans=''){
        $params = [
            'ccy' => $ccy,
            'amt' => $amt,
            'form' => $from,
            'to' => $to,
            'fromSubAccount' => $fromSubAccount,
            'toSubAccount' => $toSubAccount,
            'loanTrans' => $loanTrans,
        ];

        return $this->request('/api/v5/asset/subaccount/transfer', $params, 'POST');
    }

    public function setTransferOut($subAcct,$canTransOut=''){
        $params = [
            'subAcct' => $subAcct,
            'canTransOut' => $canTransOut,
        ];

        return $this->request('/api/v5/users/subaccount/set-transfer-out', $params, 'POST');
    }

    public function entrustSubaccountList($subAcct=''){
        $params = [
            'subAcct' => $subAcct,
        ];

        return $this->request('/api/v5/users/entrust-subaccount-list', $params, 'GET');
    }

    public function assetBalances($subAcct,$ccy=''){
        $params = [
            'subAcct' => $subAcct,
            'ccy' => $ccy,
        ];

        return $this->request('/api/v5/asset/subaccount/balances', $params, 'GET');
    }

    public function maxWithdrawal($subAcct,$ccy=''){
        $params = [
            'subAcct' => $subAcct,
            'ccy' => $ccy,
        ];

        return $this->request('/api/v5/account/subaccount/max-withdrawal', $params, 'GET');
    }

    public function managedSubaccountBills($ccy='',$type='',$subAcct='',$subUid='',$after='',$before='',$limit=''){
        $params = [
            'ccy' => $ccy,
            'type' => $type,
            'subAcct' => $subAcct,
            'subUid' => $subUid,
            'after' => $after,
            'before' => $before,
            'limit' => $limit,
        ];

        return $this->request('/api/v5/asset/subaccount/managed-subaccount-bills', $params, 'GET');
    }

    public function createSubaccount($subAcct,$type,$label,$pwd=''){
        $params = [
            'subAcct' => $subAcct,
            'type' => $type,
            'label' => $label,
            'pwd' => $pwd,
        ];

        return $this->request('/api/v5/users/subaccount/create-subaccount', $params, 'POST');
    }

    public function apikey($subAcct,$label,$passphrase,$perm='',$ip=''){
        $params = [
            'subAcct' => $subAcct,
            'label' => $label,
            'passphrase' => $passphrase,
            'perm' => $perm,
            'ip' => $ip,
        ];

        return $this->request('/api/v5/users/subaccount/apikey', $params, 'POST');
    }

    public function getApikey($subAcct,$apiKey=''){
        $params = [
            'subAcct' => $subAcct,
            'apiKey' => $apiKey,
        ];

        return $this->request('/api/v5/users/subaccount/apikey', $params, 'GET');
    }

    public function deleteApikey($subAcct,$apiKey){
        $params = [
            'subAcct' => $subAcct,
            'apiKey' => $apiKey,
        ];

        return $this->request('/api/v5/users/subaccount/delete-apikey', $params, 'POST');
    }

}

File: okex-python-sdk-api-v5/okex/subAccount_api.py
from .client import Client
from .consts import *


class SubAccountAPI(Client):
    def __init__(self, api_key, api_secret_key, passphrase, use_server_time=False, flag='1'):
        Client.__init__(self, api_key, api_secret_key, passphrase, use_server_time, flag)

    def balances(self, subAcct):
        params = {"subAcct": subAcct}
        return self._request_with_params(GET, BALANCE, params)

    def bills(self, ccy='', type='', subAcct='', after='', before='', limit=''):
        params = {"ccy": ccy, 'type': type, 'subAcct': subAcct, 'after': after, 'before': before, 'limit': limit}
        return self._request_with_params(GET, BILLs, params)

    def delete(self, pwd, subAcct, apiKey):
        params = {'pwd': pwd, 'subAcct': subAcct, 'apiKey': apiKey}
        return self._request_with_params(POST, DELETE, params)

    def reset(self, pwd, subAcct, label, apiKey, perm, ip=''):
        params = {'pwd': pwd, 'subAcct': subAcct, 'label': label, 'apiKey': apiKey, 'perm': perm, 'ip': ip}
        return self._request_with_params(POST, RESET, params)

    def create(self, pwd, subAcct, label, Passphrase, perm='', ip=''):
        params = {'pwd': pwd, 'subAcct': subAcct, 'label': label, 'Passphrase': Passphrase, 'perm': perm, 'ip': ip}
        return self._request_with_params(POST, CREATE, params)

    def view_list(self, enable='', subAcct='', after='', before='', limit=''):
        params = {'enable': enable, 'subAcct': subAcct, 'after': after, 'before': before, 'limit': limit}
        return self._request_with_params(GET, VIEW_LIST, params)

    def subAccount_transfer(self, ccy, amt, froms, to, fromSubAccount,toSubAccount):
        params = {'ccy': ccy, 'amt': amt, 'from': froms, 'to': to, 'fromSubAccount': fromSubAccount, 'toSubAccount': toSubAccount}
        return self._request_with_params(POST, SUBACCOUNT_TRANSFER, params)


